src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs

```diff
@@ -289,6 +290,11 @@ static string GenerateTestTimedOutMessage(TimeSpan maxTime)
             {
                 throw new Exception("Executing given and whens took longer than 2 minutes");
             }
+
+            if (completedTask.IsFaulted && completedTask.Exception != null)
+            {
+                ExceptionDispatchInfo.Capture(completedTask.Exception).Throw();
```

Comment:
I think in your case `throw completedTask.Exception` would be just fine since you directly rethrow